contracts/mocks/WitnetRandomnessMock.sol

```diff
@@ -0,0 +1,157 @@
+    /// @dev   ii.  a request posted in/after given block does exist, but lest than `__mockRandomizeLatencyBlocks` have elapsed.
+    /// @param _block Block number from which the search will start.
+    function getRandomnessAfter(uint256 _block)
+        public view
+        virtual override
+        returns (bytes32)
+    {
+        if (__randomize_[_block].from == address(0)) {
+            _block = getRandomnessNextBlock(_block);
+        }
+        uint256 _queryId = __randomize_[_block].witnetQueryId;
+        require(_queryId != 0, "WitnetRandomnessMock: not randomized");
+        require(block.number >= _block + __mockRandomizeLatencyBlocks, "WitnetRandomnessMock: pending randomize");
+        return blockhash(_block);
+    }
+
+    /// Mocks `true` only when a randomness request got actually posted within given block,
+    /// and at least `__mockRandomizeLatencyBlocks` have elapsed since then. 
+    function isRandomized(uint256 _block)
+        public view
+        virtual override
+        returns (bool)
+    {
+        RandomizeData storage _data = __randomize_[_block];
+        return (
+            _data.witnetQueryId != 0 
+                && block.number >= _block + __mockRandomizeLatencyBlocks
+        );
+    }
+
+    /// Mocks request to generate randomness, using underlying EVM as entropy source.
+    /// Only one randomness request per block will be actually posted. Unused funds shall 
+    /// be transfered back to the tx sender.
+    /// @dev FOR UNITARY TESTING ONLY. DO NOT USE IN PRODUCTION, AS PROVIDED RANDOMNESS
+    /// @dev WILL NEITHER BE EVM-AGNOSTIC, NOR SECURE.
+    /// @return _usedFunds Amount of funds actually used from those provided by the tx sender.
+    function randomize()
+        external payable
+        virtual override
+        returns (uint256 _usedFunds)
+    {
+        if (latestRandomizeBlock < block.number) {
+            require(
+                msg.value >= __mockRandomizeFee,
+                "WitnetRandomnessMock: reward too low"
+            );
+            // Post the Witnet Randomness request:
+            uint _queryId = ++ __mockRandomizeLatestId;
+            RandomizeData storage _data = __randomize_[block.number];
+            _data.from = msg.sender;
+            _data.witnetQueryId = _queryId;            
+            // Update block links:
+            uint256 _prevBlock = latestRandomizeBlock;
+            _data.prevBlock = _prevBlock;
+            __randomize_[_prevBlock].nextBlock = block.number;
+            latestRandomizeBlock = block.number;
+            // Throw event:
+            emit Randomized(
+                msg.sender,
+                _prevBlock,
+                _queryId,
+                witnetRandomnessRequest.hash()
+            );
+            // Transfer back unused tx value:
+            if (_usedFunds < msg.value) {
+                payable(msg.sender).transfer(msg.value - _usedFunds);
+            }
```

Comment:
This transfer only happens if this condition is met:

        if (latestRandomizeBlock < block.number) {

In the `upgradeRandomizeFee` function above there was a change to transfer back unused funds on all cases. Should that logic also apply here?

Reply:
Absolutely